### PR TITLE
add announcement for upcoming stdlib migration

### PR DESCRIPTION
**Pull request by h-vetinari**
As discussed in https://github.com/conda-forge/conda-forge.github.io/issues/2102

### CODE REVIEW

**Comment by h-vetinari** [comment]
pre-commit.ci autofix

File: news/2024-03-20-stdlib-migration.md
Comment (mfansler) [inline news/2024-03-20-stdlib-migration.md:22]: Python shouldn't receive a special callout - this equally impacts all scripting languages with compiled library interfacing (e.g., Perl, R). I think the parenthetical could be dropped altogether - "all compiled packages" is sufficiently clear. Or perhaps "(i.e., non-noarch)".

Also maybe this should be "recipes" instead of "packages.

File: news/2024-03-20-stdlib-migration.md
Comment (beckermr) [inline news/2024-03-20-stdlib-migration.md:45]: Should we put this info in our knowledgebase entries on using compilers? It might already be there, IDK.

Comment (beckermr) [inline news/2024-03-20-stdlib-migration.md:45]: >   You should then also remove any line involving `__osx` from `meta.yaml`.

Maybe it might be good to list an example of this? I don't know which lines you mean. Does this apply to say entries in run that have special osx run requirements?

Comment (jaimergp) [reply news/2024-03-20-stdlib-migration.md:45]: Agreed that this would be a welcome addition to the knowledge base or compilers. We could link to the entry from the announcement so it doesn't get as long. Should help with discoverability!

Comment (h-vetinari) [reply news/2024-03-20-stdlib-migration.md:45]: This was mainly trying to refer to (variations of) `- __osx >={{ MACOSX_DEPLOYMENT_TARGET }}  # [osx and x86_64]`

Comment (h-vetinari) [reply news/2024-03-20-stdlib-migration.md:45]: > Should we put this info in our knowledgebase entries on using compilers?

Yeah, but that would need to build on https://github.com/conda-forge/conda-forge.github.io/pull/1950, which has been waiting for feedback/resolution. Perhaps I need to bring up this topic in the core call.

> We could link to the entry from the announcement so it doesn't get as long. Should help with discoverability!

Yeah, that sounds like a good idea.

Comment (beckermr) [reply news/2024-03-20-stdlib-migration.md:45]: Got it. We should mention this explicitly.

Comment (beckermr) [reply news/2024-03-20-stdlib-migration.md:45]: That PR doesn't have the new stdlib setup stuff in it for the knowledgebase that I can see.

Comment (h-vetinari) [reply news/2024-03-20-stdlib-migration.md:45]: Yeah, that PR's about a year old. My point is that if we want to add stdlib handling to our compiler docs, I want to build on the existing clean-up rather than do two conflicting updates.

Comment (h-vetinari) [reply news/2024-03-20-stdlib-migration.md:45]: > > We could link to the entry from the announcement so it doesn't get as long. Should help with discoverability!
> 
> Yeah, that sounds like a good idea.

I actually reconsidered this and don't necessarily agree anymore. The docs definitely need to be updated, but they should describe the new status quo, not a transition (or at least not for long, as it will become stale quickly).

The announcement (or tracking issue) seems like a reasonable place for me to describe the main mechanics of the transition, especially to the degree that it allows maintainers to do it themselves, without having to wait for the piggyback migrator.

Comment (jaimergp) [reply news/2024-03-20-stdlib-migration.md:45]: My main concern was about the discoverability of that information, but the search features should reveal both, so no I have no strong feelings.

Comment (beckermr) [reply news/2024-03-20-stdlib-migration.md:45]: We should clarify this point in the PR before it is merged.

Comment (h-vetinari) [reply news/2024-03-20-stdlib-migration.md:45]: Definitely. Haven't had time to incorporate feedback here yesterday. I'm thinking to separate the announcement from the wider docs update though, which would also uncouple this from having to wait for #1950.

Comment (h-vetinari) [reply news/2024-03-20-stdlib-migration.md]: So I only added a promise to update the docs for now, for two reasons:
* I want to see this working in the wild before we write docs
* I want #1950 to land first, but the announcement shouldn't have to wait for that.

Hope that's alright.

Comment (h-vetinari) [reply news/2024-03-20-stdlib-migration.md]: Done now.

**Comment by h-vetinari** [comment]
@conda-forge/core, any opposition to merging this as-is? I'll update the docs as soon as #1950 is settled.

**Review by beckermr — APPROVED** [review]

**Review by xhochy — APPROVED** [review]

**Comment by h-vetinari** [comment]
Thanks for the reviews!

**Pull request merged**
